File: get_animations.py
import json
import struct
import sys
import glob
import os

def get_glb_animations(filepath):
    with open(filepath, 'rb') as f:
        magic = f.read(4)
        if magic != b'glTF':
            return []
        version, length = struct.unpack('<II', f.read(8))
        chunk_len, chunk_type = struct.unpack('<II', f.read(8))
        if chunk_type != 0x4E4F534A: # 'JSON'
            return []
        json_data = f.read(chunk_len).decode('utf-8')
        gltf = json.loads(json_data)
        animations = gltf.get('animations', [])
        return [anim.get('name', 'Unnamed') for anim in animations]

def main():
    directory = r'e:\webproject\OutbreakArena\assets\Models\GLB format'
    glbs = glob.glob(os.path.join(directory, 'character-*.glb'))
    for glb in glbs:
        name = os.path.basename(glb)
        anims = get_glb_animations(glb)
        print(f"{name}: {anims}")

if __name__ == '__main__':
    main()

File: get_animations.js
const fs = require('fs');
const path = require('path');

function getGlbAnimations(filepath) {
    const buffer = fs.readFileSync(filepath);
    const magic = buffer.toString('utf8', 0, 4);
    if (magic !== 'glTF') return [];
    
    // const version = buffer.readUInt32LE(4);
    // const length = buffer.readUInt32LE(8);
    
    const chunkLen = buffer.readUInt32LE(12);
    const chunkType = buffer.readUInt32LE(16);
    
    if (chunkType !== 0x4E4F534A) return []; // 'JSON'
    
    const jsonData = buffer.toString('utf8', 20, 20 + chunkLen);
    const gltf = JSON.parse(jsonData);
    
    const animations = gltf.animations || [];
    return animations.map(a => a.name || 'Unnamed');
}

const dir = 'e:\\webproject\\OutbreakArena\\assets\\Models\\GLB format';
const files = fs.readdirSync(dir);
const characters = files.filter(f => f.startsWith('character-') && f.endsWith('.glb'));

characters.forEach(char => {
    const anims = getGlbAnimations(path.join(dir, char));
    console.log(`${char}: ${anims.join(', ')}`);
});
